### PR TITLE
Added mssql_user state

### PR DESCRIPTION
**Pull request by pakdel**
### What does this PR do?
Add `mssql_user.present` and `mssql_user.absent`
It also adds minor features to `mssql` execution module

### Tests written?
No

### CODE REVIEW

File: salt/modules/mssql.py [hunk 1]
```diff
@@ -250,23 +252,69 @@ def login_exists(login, **kwargs):
...
+    '''
+    Find if an user exists in a specific database on the MS SQL server.
 
     CLI Example:
 
     .. code-block:: bash
 
         salt minion mssql.user_exists 'USERNAME' [database='DBNAME']
     '''
-    # 'database' argument is mandatory
```
Comment (pakdel) [inline salt/modules/mssql.py]: 'database' has a default value and is not mandatory

File: salt/modules/mssql.py [hunk 2]
```diff
@@ -250,23 +252,69 @@ def login_exists(login, **kwargs):
         return 'Could not find the login: {0}'.format(e)
 
 
-def user_exists(username, **kwargs):
+def login_create(login, new_login_password=None, new_login_domain=None, new_login_options=[], **kwargs):
```
Comment (cachedout) [inline salt/modules/mssql.py]: We cannot have mutable values like lists as default keyword arguments. Please change this to something like `None` and then transform it to a list inside the function if necessary.

File: salt/modules/mssql.py [hunk 3]
```diff
@@ -284,42 +332,52 @@ def user_list(**kwargs):
     return [row[0] for row in tsql_query("SELECT name FROM sysusers where issqluser=1 or isntuser=1", as_dict=False, **kwargs)]
 
 
-def user_create(username, new_login_password=None, **kwargs):
+def user_create(username, login=None, domain=None, database=None, roles=[], options=[], **kwargs):
```
Comment (cachedout) [inline salt/modules/mssql.py]: Same here. Please don't use lists as defaults.

File: salt/states/mssql_user.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    if type(options) in [dict, collections.OrderedDict]:
+        return [ '{0}={1}'.format(k, v) for k, v in options.items() ]
+    if type(options) is list and (not len(options) or type(options[0]) is str):
+        return options
+    # Invalid options
+    if type(options) is not list or type(options[0]) not in [dict, collections.OrderedDict]:
+        return []
+    return [ o for d in options for o in _normalize_options(d) ]
+
+def present(name, password=None, domain=None, database=None, roles=[], login_options=[], user_options=[], **kwargs):
```
Comment (cachedout) [inline salt/states/mssql_user.py]: Same here. Please no default lists.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by pakdel** [comment]
@cachedout I believe I fixed those issues

**Comment by pakdel** [comment]
Deprecated by https://github.com/saltstack/salt/pull/42460

**Pull request closed without merging**